04.ls/ls.rb

```diff
@@ -1,27 +1,101 @@
 # frozen_string_literal: true
 
 require 'optparse'
+require 'etc'
+require 'date'
+require 'pathname'
 
 def main
   opt = OptionParser.new
   params = {}
-  opt.on('-r') { |v| params[:r] = v }
+  opt.on('-l') { |v| params[:l] = v }
   opt.parse!(ARGV)
-
   files = fetch_filenames_without_dotfile
-  files = files.reverse if params[:r]
-  output_list(files)
+
+  if params[:l]
+    puts show_file_detail(files)
+  else
+    output_list(files)
+  end
 end
 
 def fetch_filenames_without_dotfile
   Dir.glob('*')
 end
 
+def show_file_detail(files)
+  detail_lines = []
+  total_block_size = 0
+  files.each do |file|
+    path = File.expand_path(file)
+    total_block_size += get_file_stat(path).blocks
+
+    line = [change_file_permission_format(path), count_hardlink(path), find_owner_name(path), find_group_name(path), calculate_bytesize(path).rjust(4),
+            export_timestamp(path), file].join(' ')
```

Comment:
ちょっと横に長くて読みづらいので、改行することをお勧めします。
```suggestion
    line = [
      change_file_permission_format(path),
      count_hardlink(path),
      find_owner_name(path),
      find_group_name(path),
      calculate_bytesize(path).rjust(4),
      export_timestamp(path),
      file
    ].join(' ')
```